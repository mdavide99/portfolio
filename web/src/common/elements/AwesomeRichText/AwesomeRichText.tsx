import classNames from "classnames";
import BlockContent from "@sanity/block-content-to-react";

interface AwesomeRichTextProps {
  content: JSX.Element;
}

const serializers = {
  marks: {
    link: ({
      mark,
      children,
    }: {
      mark: { blank: string; href: string };
      children: Element;
    }) => {
      const linkClass = classNames(
        "underline decoration-2 decoration-blue-500 dark:decoration-fuchsia-500 underline-offset-4 hover:underline-offset-2"
      );
      const { href } = mark;
      return (
        <a
          href={href}
          target="_blank"
          rel="noopener noreferrer"
          className={linkClass}
        >
          {children}
        </a>
      );
    },
    tec: ({ children }: { children: Element }) => {
      const linkClass = classNames("font-light");
      return <span className={linkClass}>{children}</span>;
    },
  },
};

function AwesomeRichText(props: AwesomeRichTextProps) {
  return <BlockContent blocks={props.content} serializers={serializers} />;
}

export { AwesomeRichText };
