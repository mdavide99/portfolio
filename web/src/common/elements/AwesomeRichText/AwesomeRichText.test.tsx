import { AwesomeRichText } from "./AwesomeRichText";
import { render } from "@testing-library/react";

describe("AwesomeRichText", () => {
  it("renders AwesomeRichText correctly", () => {
    const { container } = render(<AwesomeRichText content={<p>TEST</p>} />);
    expect(container).toMatchSnapshot();
  });
});
