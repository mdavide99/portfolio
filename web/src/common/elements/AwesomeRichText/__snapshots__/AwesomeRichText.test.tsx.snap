// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AwesomeRichText renders AwesomeRichText correctly 1`] = `
<div>
  <div
    style="display: none;"
  >
    Unknown block type "undefined", please specify a serializer for it in the \`serializers.types\` prop
  </div>
</div>
`;
